pkg/agent/flowexporter/connections/connections.go

```diff
@@ -122,16 +120,23 @@ func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
 	cs.mutex.Lock()
 	defer cs.mutex.Unlock()
 	if exists {
+		// avoid updating stats of the existing connection that is about to close
+		if flowexporter.IsConnectionDying(existingConn) {
+			existingConn.IsPresent = true
+			cs.connections[connKey] = existingConn
+			return
+		}
 		// Update the necessary fields that are used in generating flow records.
 		// Can same 5-tuple flow get deleted and added to conntrack table? If so use ID.
 		existingConn.StopTime = conn.StopTime
 		existingConn.OriginalBytes = conn.OriginalBytes
 		existingConn.OriginalPackets = conn.OriginalPackets
 		existingConn.ReverseBytes = conn.ReverseBytes
 		existingConn.ReversePackets = conn.ReversePackets
+		existingConn.TCPState = conn.TCPState
 		existingConn.IsPresent = true
 		// Reassign the flow to update the map
-		cs.connections[connKey] = *existingConn
+		cs.connections[connKey] = existingConn
```

Comment:
Do we need this? Now we are just updating the pointer directly.

Reply:
I kept these re-assignments of connections here and in `flow_records.go` when updating flow records so that we can detect the TCP state change [here](https://github.com/vmware-tanzu/antrea/blob/b5ab9783d004ec76807d6f7338db2548f73c64e5/pkg/agent/flowexporter/flowrecords/flow_records.go#L70).

Reply:
I agree with @srikartati I don't understand this part.
It seems that this pointer assignment is a no-op. Also we need to remove the call to `cs.GetConnByKey(connKey)` above IMO.

Instead of:
```go
	existingConn, exists := cs.GetConnByKey(connKey)
	cs.mutex.Lock()
	defer cs.mutex.Unlock()
```
the code should be:
```go
	cs.mutex.Lock()
	defer cs.mutex.Unlock()
         existingConn, exists := cs.connections[connKey]
```

The lock should not be released then re-acquired

Reply:
Just realized my misunderstanding of pointer re-assignment. Added `TCPState` field to records for detecting tcp state change. Thanks!